src/mainwindow.cpp

```diff
@@ -337,3 +340,19 @@ void MainWindow::publishTools(GraphicMap::GraphicLayer* layer) {
     auto toolbox = ui->widgetDrawingToolbox;
     toolbox->reset(m_mapScene, m_chipsetScene, tools);
 }
+
+void MainWindow::closeEvent (QCloseEvent *event)
+{
+    QMessageBox::StandardButton resBtn = QMessageBox::question( this, "APP_NAME",
```

Comment:
Replace `APP_NAME` with something revelant.

Also, please make sure that lines do not exceed 80 columns, linefeed included (I’ll explain if you don't understand, this might be a bit odd when you get started).